### PR TITLE
Integrate SetFit with API Inference + Tests

### PR DESCRIPTION
**Pull request by tomaarsen**
Hello!

## Pull Request overview
* Integrate [SetFit](https://github.com/huggingface/setfit) into API Inference
  * [x] Copied `docker_images/common`.
  * [x] Edited `docker_images/setfit/requirements.txt`, `docker_images/setfit/main.py` and `docker_images/setfit/pipelines/token_classification.py`.
  * [x] Removed unused pipeline files, tests and imports.
  * [x] Edited `setfit/tests/test_api.py` with a `MiniLM` model for quick tests.
  * [x] Edited `tests/test_dockers.py` with a new test for `def test_setfit(self)`.
  * [x] Added two workflows (`python-api-setfit-cd.yaml` and `python-api-setfit.yaml`)

## Details
[SetFit](https://github.com/huggingface/setfit) is a library for text classification with ~1200 models on the Hub at the time of writing. A v1.0.0 release is upcoming, and it's a good time to add this widget support. 

I've used my [`tomaarsen/setfit-all-MiniLM-L6-v2-sst2-32-shot`](https://huggingface.co/tomaarsen/setfit-all-MiniLM-L6-v2-sst2-32-shot) model throughout the tests. This model is based on [`sentence-transformers/all-MiniLM-L6-v2`](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) embedding model, which should be fairly small (~90MB).

To the best of my knowledge, I've followed all of the steps in the README and from the [integration documentation](https://huggingface.co/docs/hub/models-adding-libraries). Please let me know if you need anything else from me at this point!

Presumably I don't need to mess around with https://github.com/huggingface/api-inference-community/pull/158?

Related PRs:
* https://github.com/huggingface/hub-docs/pull/1150
* https://github.com/huggingface/huggingface.js/pull/387

- Tom Aarsen

### CODE REVIEW

File: docker_images/setfit/app/pipelines/text_classification.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ) -> None:
+        self.model = SetFitModel.from_pretrained(model_id)
+
+    def __call__(self, inputs: str) -> List[Dict[str, float]]:
+        """
+        Args:
+            inputs (:obj:`str`):
+                a string containing some text
+        Return:
+            A :obj:`list`:. The object returned should be a list of one list like [[{"label": 0.9939950108528137}]] containing:
```
Comment (osanseviero) [inline docker_images/setfit/app/pipelines/text_classification.py:20]:
```suggestion
            A :obj:`list`: The object returned should be a list of one list like [[{"label": 0.9939950108528137}]] containing:
```

File: docker_images/setfit/app/pipelines/text_classification.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                - "label": A string representing what the label/class is. There can be multiple labels.
+                - "score": A score between 0 and 1 describing how confident the model is for this label/class.
+        """
+        probs = self.model.predict_proba([inputs], as_numpy=True)
+        if probs.ndim == 2:
+            id2label = {}
+            if hasattr(self.model, "id2label"):
+                id2label = self.model.id2label
```
Comment (osanseviero) [inline docker_images/setfit/app/pipelines/text_classification.py:28]: Or

```suggestion
            getattr(self.model, "id2label", {})
```

File: docker_images/setfit/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+starlette==0.27.0
+api-inference-community==0.0.32
+huggingface_hub==0.11.0
+setfit
```
Comment (osanseviero) [inline docker_images/setfit/requirements.txt:4]: If possible, it's a good idea to pin a setfit version so we know exactly what's in production and we're reliable upon rebuilds

File: docker_images/setfit/tests/test_api_text_classification.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+from app.main import ALLOWED_TASKS
+from starlette.testclient import TestClient
+from tests.test_api import TESTABLE_MODELS
+
+
+@skipIf(
+    "text-classification" not in ALLOWED_TASKS,
+    "text-classification not implemented",
+)
+class TextClassificationTestCase(TestCase):
```
Comment (osanseviero) [inline docker_images/setfit/tests/test_api_text_classification.py:14]: Just to confirm; did you run these tests?

**Review by osanseviero — APPROVED** [review]
Looks great, thanks!

**Review by Narsil — APPROVED** [review]
LGTM.

**Pull request merged**